### DISCUSSION TITLE
add another project beside project of serenity

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5365] [Help]
please help meeeeeeeeeeee

now i want to make the web-site it self which i build their admin panel by using serenity 
my question is how to take all data from serenity without need to retrieve the data from database again ?
and another question do i need to create mvc project (the new project which contain the the interfaces ) or its ok if i make it forms (web form)?

**Comment by VictorTomaili** [comment]
You can create standart mvc pages in your serenity project. There is no restriction about that. Same project for end users and admin user is ok. 

If you want to create separated project for end user, best way is get data from data source (sql, redis, mongodb etc.) and use cache strategies there. Don't try to get data from admin panel via api's because it's bring web traffic to your admin panel either. It's mean when you get any load issue, you also can't access your admin panel. Also when you need update your admin panel, it's mean your end user project also be effected about that.

**Reply by samaremad** [reply]
ok how can i create pages for end users in same serenity project so the users can access the homepage only while the admin can access the admin panel and the home page how can i do that ?

**Reply by erehtolleh** [reply]
Not sure we understand your requirements.  User can only access the homepage when the admin have access?

**Reply by samaremad** [reply]
no i want to create homepage by sereinty project without need to create a new mvc project

**Reply by erehtolleh** [reply]
Did you look at the sample project?  Like the Dashboard.  That gives you a pretty good idea on how to make a separate page within serenity.  You have to create the controller and the cshtml page.

I would create a sample Serenity project with all the samples loaded and run it, so you can see it in action.

**Reply by samaremad** [reply]
when you would create that ? 
can i get contact with you to understand well from you so that you can help me with my project 
it is so urgent

**Reply by erehtolleh** [reply]
look at the sample page for the dashboard.  It shows you how you can create a custom page.

**Reply by samaremad** [reply]
can you give me the link i didnt get you well which page do you mean ?
